### PR TITLE
[thread-cert] add case 7.1.8 BorderRouterAsFED using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        #         message with the server’s information (Prefix, Border Router)
+        #         to the Leader.
+        pkts.filter_wpan_src64(FED).\
+            filter_ipv6_dst(LEADER_ALOC).\
+            filter_coap_request(SVR_DATA_URI).\
+            filter(lambda p: {
+                              Ipv6Addr(PREFIX_2001[:-3]),
+                              Ipv6Addr(PREFIX_2002[:-3])
+                             } == set(p.thread_nwd.tlv.prefix) and\
+                   p.thread_nwd.tlv.border_router_16 is not nullField
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:151]: can we check the border router RLOC16 ?

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                   ).\
+            must_next()
+
+        # Step 3: Leader transmits a 2.04 Changed CoAP response to the DUT.
+        #         And transmits multicast MLE Data Response with the new information
+        #         collected, adding also 6LoWPAN ID TLV for the prefix set on FED.
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:157]: have we checked `6LoWPAN ID TLV`?

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                       p.thread_nwd.tlv.border_router.flag.p == [1, 1] and\
+                       p.thread_nwd.tlv.border_router.flag.s == [1, 1] and\
+                       p.thread_nwd.tlv.border_router.flag.r == [1, 1] and\
+                       p.thread_nwd.tlv.border_router.flag.o == [1, 1] and\
+                       p.thread_nwd.tlv.stable == [0, 1, 1, 1, 0, 0, 0]
+                       ).\
+                must_next()
+
+        # Step 4: The DUT transmits multicast MLE Data Response with the new information
+        #         collected, adding also 6LoWPAN ID TLV for the prefix set on FED.
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:180]: have we checked `6LoWPAN ID TLV` ?

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                   p.thread_nwd.tlv.border_router.flag.s == [1, 1] and\
+                   p.thread_nwd.tlv.border_router.flag.r == [1, 1] and\
+                   p.thread_nwd.tlv.border_router.flag.o == [1, 1] and\
+                   p.thread_nwd.tlv.stable == [0, 1, 1, 1, 0, 0, 0]
+                   ).\
+            must_next()
+
+        # Step 6: The DUT notifies Leader of removed server’s (FED’s) RLOC16.
+        #         The DUT MUST send a CoAP Server Data Notification frame
+        #         to the Leader containing only the removed server’s RLOC16:
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:198]: need check that the TLVs `containing only the removed server’s RLOC16` without other TLVs?

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py (hunk 3 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:183]: added

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py (hunk 4 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:204]: added the check

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py]: added the check

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py (hunk 2 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:157]: added

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+            pkts.filter_wpan_src64(LEADER).\
+                filter_LLANMA().\
+                filter_mle_cmd(MLE_DATA_RESPONSE).\
+                filter(lambda p: {
+                                  Ipv6Addr(PREFIX_2001[:-3]),
+                                  Ipv6Addr(PREFIX_2002[:-3])
+                                 } == set(p.thread_nwd.tlv.prefix) and\
+                                {
+                                 NWD_6LOWPAN_ID_TLV
+                                 } <= set(p.thread_nwd.tlv.type) and\
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:173]:
```suggestion
                                NWD_6LOWPAN_ID_TLV in p.thread_nwd.tlv.type and\
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
with a nit

File: tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py (hunk 5 above)
Comment (jwhui) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py:173]: @FiveDimensions , are you planning to address this?

Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_08_BorderRouterAsFED.py]: updated, thanks

**Review by jwhui — APPROVED** [review]

**Pull request merged**